homeassistant/components/sensor/eight_sleep.py

```diff
@@ -0,0 +1,189 @@
+        state_attr[ATTR_ACTIVE_HEAT] = self._usrobj.now_heating
+        state_attr[ATTR_DURATION_HEAT] = self._usrobj.heating_remaining
+        state_attr[ATTR_LAST_SEEN] = self._usrobj.last_seen
+
+        return state_attr
+
+
+class EightUserSensor(EightSleepUserEntity):
+    """Representation of a eight sleep user-based sensor."""
+
+    def __init__(self, name, eight, sensor, units):
+        """Initialize the sensor."""
+        super().__init__(eight)
+
+        self._sensor = sensor
+        self._sensor_root = self._sensor.split('_', 1)[1]
+        self._mapped_name = NAME_MAP.get(self._sensor, self._sensor)
+        self._name = '{} {}'.format(name, self._mapped_name)
+        self._state = None
+        self._attr = None
+        self._units = units
+
+        self._side = self._sensor.split('_', 1)[0]
+        self._userid = self._eight.fetch_userid(self._side)
+        self._usrobj = self._eight.users[self._userid]
+
+        _LOGGER.debug('User Sensor: %s, Side: %s, User: %s',
+                      self._sensor, self._side, self._userid)
+
+    @property
+    def name(self):
+        """Return the name of the sensor, if any."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Retrieve latest state."""
+        _LOGGER.debug('Updating User sensor: %s', self._sensor)
+        if 'current' in self._sensor_root:
+            self._state = self._usrobj.current_sleep_score
+            self._attr = self._usrobj.current_values
+        elif 'last' in self._sensor_root:
+            self._state = self._usrobj.last_sleep_score
+            self._attr = self._usrobj.last_values
+
+    @property
+    def device_state_attributes(self):
+        """Return device state attributes."""
+        state_attr = {ATTR_SESSION_START: self._attr['date']}
+        state_attr[ATTR_TNT] = self._attr['tnt']
+        state_attr[ATTR_PROCESSING] = self._attr['processing']
+
+        sleep_time = sum(self._attr['breakdown'].values()) - \
+            self._attr['breakdown']['awake']
+        state_attr[ATTR_SLEEP_DUR] = sleep_time
+        state_attr[ATTR_LIGHT_PERC] = round((
+            self._attr['breakdown']['light'] / sleep_time) * 100, 2)
+        state_attr[ATTR_DEEP_PERC] = round((
+            self._attr['breakdown']['deep'] / sleep_time) * 100, 2)
+
+        if self._units == 'si':
+            room_temp = round(self._attr['room_temp'], 2)
+            bed_temp = round(self._attr['bed_temp'], 2)
+        else:
+            room_temp = round((self._attr['room_temp']*1.8)+32, 2)
+            bed_temp = round((self._attr['bed_temp']*1.8)+32, 2)
+
+        if 'current' in self._sensor_root:
+            state_attr[ATTR_RESP_RATE] = round(self._attr['resp_rate'], 2)
+            state_attr[ATTR_HEART_RATE] = round(self._attr['heart_rate'], 2)
```

Comment:
I can go that route if you think it makes more sense.  My intention was to make the data available and let the user template out whichever values they were interested in since they are only updated when a sleep session is active.  Right now we're at 8 total sensors for both left/right sides of the bed, if I split out the active session values we'd be adding another 10.  It just seemed like data overload to me.